### PR TITLE
[STK-368] Add api implementation of forecastle backend in go

### PR DESCRIPTION
**Pull request by waseem-h**
What's done in this PR:
- [x] Remove nodejs dependency completely
- [x] Serve frontend statically on / in go
- [x] Clean up dockerfile, jenkinsfile etc and make it a pure go project
- [x] Write API test cases
- [x] Add logging to API
- [x] Update frontend to use that API
- [x] Add frontend magic to group apps per namespaces
- [x] Update Readme and screenshot according to the changes made by this PR, and also version (if required to bump major)

### CODE REVIEW

**Comment by stakater-user** [comment]
@waseem-h Image is available for testing. ``docker pull stakater/forecastle:PR-32-1``

**Comment by stakater-user** [comment]
@waseem-h Image is available for testing. ``docker pull stakater/forecastle:PR-32-2``

**Comment by stakater-user** [comment]
@waseem-h Image is available for testing. ``docker pull stakater/forecastle:PR-32-3``

**Comment by stakater-user** [comment]
@waseem-h Image is available for testing. ``docker pull stakater/forecastle:PR-32-4``

**Comment by stakater-user** [comment]
@waseem-h Yikes! You better fix it before anyone else finds out! [Build 5](https://jenkins.tools.stackator.com/job/Stakater/job/Forecastle/job/PR-32/5/) has Failed!

**Comment by stakater-user** [comment]
@waseem-h Image is available for testing. ``docker pull stakater/forecastle:PR-32-6``

**Comment by stakater-user** [comment]
@waseem-h Image is available for testing. ``docker pull stakater/forecastle:PR-32-7``

**Comment by stakater-user** [comment]
@waseem-h Image is available for testing. ``docker pull stakater/forecastle:PR-32-9``

**Comment by stakater-user** [comment]
@waseem-h Image is available for testing. ``docker pull stakater/forecastle:PR-32-10``

File: pkg/log/log.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	filenameHook := filename.NewHook()
+
+	log := &logrus.Logger{
+		Hooks:     make(logrus.LevelHooks),
+		Out:       os.Stdout,
+		Formatter: &logrus.TextFormatter{},
+		Level:     logrus.InfoLevel,
+	}
+
+	log.Hooks.Add(filenameHook)
```
Comment (hazim1093) [inline pkg/log/log.go]: Add comment why we needed this abstraction and what this line adds to the log

**Comment by hazim1093** [comment]
Implements https://github.com/stakater/Forecastle/issues/24

**Comment by stakater-user** [comment]
@waseem-h Image is available for testing. ``docker pull stakater/forecastle:PR-32-11``

**Comment by stakater-user** [comment]
@waseem-h Image is available for testing. ``docker pull stakater/forecastle:PR-32-12``

**Comment by stakater-user** [comment]
@waseem-h Image is available for testing. ``docker pull stakater/forecastle:PR-32-13``

**Comment by stakater-user** [comment]
@waseem-h Image is available for testing. ``docker pull stakater/forecastle:PR-32-14``

**Comment by stakater-user** [comment]
@waseem-h Image is available for testing. ``docker pull stakater/forecastle:PR-32-15``

**Pull request merged**